src/ResourceManager/ServiceFabric/Commands.ServiceFabric/Commands/ServiceFabricClusterCmdlet.cs

```diff
@@ -0,0 +1,144 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Management.Automation;
+using System.Threading.Tasks;
+using Microsoft.Azure.Commands.ServiceFabric.Models;
+using Microsoft.Azure.Management.ServiceFabric;
+using Microsoft.Azure.Management.ServiceFabric.Models;
+using Newtonsoft.Json.Linq;
+using ServiceFabricProperties = Microsoft.Azure.Commands.ServiceFabric.Properties;
+
+namespace Microsoft.Azure.Commands.ServiceFabric.Commands
+{
+    public abstract class ServiceFabricClusterCmdlet : ServiceFabricCmdletBase
+    {
+        [Parameter(Mandatory = true, Position = 1, ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Specify the name of the cluster")]
+        [ValidateNotNullOrEmpty()]
+        public virtual string ClusterName { get; set; }
```

Comment:
Should be called Name, or always have an alias to Name - this allows for piping from the generic resource cmdlets